src/Container.ts

```diff
@@ -40,38 +49,42 @@ export class Container {
     /**
      * Registers a new service.
      *
-     * @param name Service name. Optional
-     * @param type Service class
-     * @param params Parameters to be sent to constructor on service initialization
+     * @param descriptor
      */
-    static registerService(name: string, type: Function, params?: any[]) {
-        this.registeredServices.push({ name: name, type: type, params: params });
+    static registerService(descriptor: ServiceDescriptor) {
+        this.registeredServices.push(descriptor);
     }
 
     /**
      * Retrieves the service with the specific name or given type from the service container.
-     * Optionally parameters can be pass in the case if instance is initialized in the container for the first time.
+     * Optionally, parameters can be passed in case if instance is initialized in the container for the first time.
      */
     static get<T>(type: ConstructorFunction<T>, params?: any[]): T;
     static get<T>(name: string, params?: any[]): T;
     static get<T>(typeOrName: ConstructorFunction<T>|string, params?: any[]): T {
 
         // normalize parameters
-        let type: Function, name: string;
+        let type: Function;
+        let name: string;
+        let factory: Function;
```

Comment:
The human eye works in "F" pattern (there are studies), i.e. it scans lines vertically first, then it scans the interesting lines horizontally. So it's better to keep important stuff closer to the start of the line and to split it into multiple lines. For me it's much easier to find `name` variable in this case when it's defined on it's own line. Also, when each variable has it's own definition, we can quickly change it's type from `const` to `let` for example, otherwise we will have to split them beforehand.